src/shared/tpm2-util.c

```diff
@@ -1487,6 +1487,73 @@ static int tpm2_get_policy_digest(
         return 0;
 }
 
+static int tpm2_create(
+                Tpm2Context *c,
+                const Tpm2Handle *parent,
+                const Tpm2Handle *session,
+                const TPMT_PUBLIC *template,
+                const TPMS_SENSITIVE_CREATE *sensitive,
+                TPM2B_PUBLIC **ret_public,
+                TPM2B_PRIVATE **ret_private) {
+
+        usec_t ts;
+        TSS2_RC rc;
+
+        assert(c);
+        assert(template);
+
+        log_debug("Creating object on TPM.");
+
+        ts = now(CLOCK_MONOTONIC);
+
+        TPM2B_PUBLIC tpm2b_public = {
+                .size = sizeof(*template),
+                .publicArea = *template,
+        };
+
+        /* Zero the unique area. */
+        zero(tpm2b_public.publicArea.unique);
+        tpm2b_public.size -= sizeof(tpm2b_public.publicArea.unique);
+
+        TPM2B_SENSITIVE_CREATE tpm2b_sensitive = {};
+        if (sensitive) {
+                tpm2b_sensitive.size = sizeof(*sensitive);
+                tpm2b_sensitive.sensitive = *sensitive;
+        }
```

Comment:
maybe:

```c
TPM2B_SENSITIVE_CREATE tpm2b_sensitive;
if (sensitive) 
        *tpm2b_sensitive = (TPM2B_SENSITIVE_CREATE) {
                .size = sizeof(*sensitive);
                .sensitive = *sensitive;
        };
else
        *tpm2b_sensitive = (TPM2B_SENSITIVE_CREATE) {};
```
or so? but doesn't matter...

Reply:
updated, but left the else {} out, instead i updated the Esys call to pass `NULL` if `sensitive` is `NULL`.

Reply:
actually i think tpm2-tss will segfault if sensitive param is null, so i updated this to exactly as you suggested (with else)